pkgs/applications/graphics/ktikz/default.nix

```diff
@@ -46,24 +53,22 @@ stdenv.mkDerivation rec {
     # lrelease command:
     LRELEASECOMMAND = lrelease
     # qcollectiongenerator command:
-    #QCOLLECTIONGENERATORCOMMAND = qcollectiongenerator
+    QCOLLECTIONGENERATORCOMMAND = qhelpgenerator
 
     # TikZ documentation default file path:
     TIKZ_DOCUMENTATION_DEFAULT = @out@/share/doc/texmf/pgf/pgfmanual.pdf.gz
```

Comment:
```suggestion
    TIKZ_DOCUMENTATION_DEFAULT = ${pgf}/share/texmf-nix/doc/generic/pgf/pgfmanual.pdf
```

Reply:
Actually, `qmakeFlags` should be sufficient.

```patch
diff --git a/pkgs/applications/graphics/ktikz/default.nix b/pkgs/applications/graphics/ktikz/default.nix
index 1f3d094e9da..c7cc9d89a4c 100644
--- a/pkgs/applications/graphics/ktikz/default.nix
+++ b/pkgs/applications/graphics/ktikz/default.nix
@@ -1,5 +1,5 @@
 { stdenv, fetchFromGitHub, fetchurl
-, gettext, poppler, qt5 , pkgconfig }:
+, gettext, poppler, qt5, pkgconfig, pgf }:
 
 # Warning: You will also need a working pdflatex installation containing
 # at least auctex and pgf.
@@ -32,43 +32,14 @@
     maintainers = [ maintainers.layus ];
   };
 
-  conf = ''
-    # installation prefix:
-    PREFIX = @out@
+  qmakeFlags = [
+    "DESKTOP_INSTALL_DIR=${placeholder "out"}/share/applications"
+    "MIME_INSTALL_DIR=${placeholder "out"}/share/mime/packages"
+    "QCOLLECTIONGENERATORCOMMAND=qhelpgenerator"
+    "TIKZ_DOCUMENTATION_DEFAULT=${pgf}/share/texmf-nix/doc/generic/pgf/pgfmanual.pdf"
+  ];
 
-    # install desktop file here (*nix only):
-    DESKTOP_INSTALL_DIR = @out@/share/applications
-
-    # install mimetype here:
-    MIME_INSTALL_DIR = @out@/share/mime/packages
-
-    # install doc here:
-    MAN_INSTALL_DIR = @out@/share/man
-
-    CONFIG -= debug
-    CONFIG += release
-
-    # qmake command:
-    QMAKECOMMAND = qmake
-    # lrelease command:
-    LRELEASECOMMAND = lrelease
-    # qcollectiongenerator command:
-    QCOLLECTIONGENERATORCOMMAND = qhelpgenerator
-
-    # TikZ documentation default file path:
-    TIKZ_DOCUMENTATION_DEFAULT = @out@/share/doc/texmf/pgf/pgfmanual.pdf.gz
-  '';
-
-  # 1. Configuration is done by overwriting qtikzconfig.pri
-  postPatch = ''
-    echo "$conf" | sed "s!@out@!$out!g" > qmake/qtikzconfig.pri
-  '';
-
-  configurePhase = ''
-      qmake PREFIX="$out" ./qtikz.pro
-  '';
-
-  nativeBuildInputs = [ pkgconfig qt5.qttools ];
+  nativeBuildInputs = [ pkgconfig qt5.qttools qt5.qmake ];
   buildInputs = [ gettext qt5.full poppler ];
 
   enableParallelBuilding = true;
```

Reply:
@jtojnar Thanks! This does indeed produce a valid package. Are your sure that it also sets `CONFIG += release` correctly ?

I could find no clue about this.

Reply:
The [original file](https://github.com/fhackenberger/ktikz/blob/d7f983be538c8f3e6bd678a600208552326c6c4c/qmake/qtikzconfig.pri#L45-L46) contains the same lines so not rewriting it should not change anything.

Reply:
Well, then the value for TIKZ_DOCUMENTATION_DEFAULT (`"/usr/share/doc/texmf/pgf/pgfmanual.pdf.gz"`) from that file will also take precedence over anything we could set from qmakeflags, right ?

Reply:
I think `qmakeFlags` override the `.pri` files just like `makeFlags` override `Makefile`s.

Reply:
In this case it does not change much. I have tested with this flag set, and without. In general, qtikz manages to get himself lost by looking in strange locations, despite that setting. So I guess the user would have better chances at getting a valid texlive install before consulting the manual.

This package is funny, because it is tex-related but has no dependency on any tex packages ;-)

Reply:
That is weird. The other `qmakeFlags` appear to work, don’t they?

Reply:
Even if the includes overrode the `qmakeFlags` https://github.com/fhackenberger/ktikz/blob/d7f983be538c8f3e6bd678a600208552326c6c4c/qtikz.pro#L10-L12, why does it use `$PREFIX` instead of `/usr/share`?

Reply:
I would just patch it in the file with `substituteAll` now and report upstream the need to be able to set it as `qmake` argument:

https://github.com/NixOS/nixpkgs/blob/c68b82c7795b942415ac5bdfc5e485ee71bf8b70/pkgs/development/libraries/qt-5/hooks/qmake-hook.sh#L4-L11

Also we apparently already set `PREFIX`.

Reply:
Well, it is more complicated than that, as the value is only a default used when everything else fails. I see no point in packaging pgf manual when the whole pdflatex suite is not installed. And when installed, the "correct" manual is taken from the latex path.

Ignoring this for now and removing any patching of TIKZ_DOCUMENTATION_DEFAULT.